Compile ELM as JSON not working

It appears that since the upgrade to the 4.0 eclipse dependencies ~ cql version 3.4, the json compiling stopped working. Here is the stack:

```
Exception in thread "main" java.lang.RuntimeException: jakarta.xml.bind.PropertyException: name: eclipselink.media-type value: application/json
        at org.cqframework.cql.elm.serializing.jaxb.ElmJsonLibraryWriter.write(ElmJsonLibraryWriter.java:26)
        at org.cqframework.cql.cql2elm.CqlTranslator.convertToJson(CqlTranslator.java:208)
        at org.cqframework.cql.cql2elm.CqlTranslator.toJson(CqlTranslator.java:130)
        at org.cqframework.cql.cql2elm.CqlTranslator.toJson(CqlTranslator.java:141)
        at org.cqframework.cql.cql2elm.cli.Main.writeELM(Main.java:100)
        at org.cqframework.cql.cql2elm.cli.Main.main(Main.java:243)
Caused by: jakarta.xml.bind.PropertyException: name: eclipselink.media-type value: application/json
        at jakarta.xml.bind.helpers.AbstractMarshallerImpl.setProperty(AbstractMarshallerImpl.java:352)
        at org.glassfish.jaxb.runtime.v2.runtime.MarshallerImpl.setProperty(MarshallerImpl.java:492)
        at org.cqframework.cql.elm.serializing.jaxb.ElmJsonLibraryWriter.write(ElmJsonLibraryWriter.java:24)
        ... 5 more
```


I believe I narrowed this down to a newer implementation added `org.eclipse.persistence.jaxb.XMLBindingContextFactory`

The following is a workaround which might help identify the root of the issue
`java -Djakarta.xml.bind.JAXBContextFactory=org.eclipse.persistence.jaxb.XMLBindingContextFactory -classpath .... org.cqframework.cql.cql2elm.cli.Main`

--- a/Src/java/.vscode/settings.json
+++ b/Src/java/.vscode/settings.json
@@ -30,7 +30,8 @@
         "tngtech",
         "trackback",
         "Ucum",
-        "unescaping"
+        "unescaping",
+        "Unmarshaller"
     ],
     "cSpell.enabledLanguageIds": [
         "java",

--- a/Src/java/elm-jaxb/src/main/java/org/cqframework/cql/elm/serializing/jaxb/ElmJsonMapper.java
+++ b/Src/java/elm-jaxb/src/main/java/org/cqframework/cql/elm/serializing/jaxb/ElmJsonMapper.java
@@ -2,17 +2,23 @@ package org.cqframework.cql.elm.serializing.jaxb;
 
 import jakarta.xml.bind.JAXBContext;
 import jakarta.xml.bind.JAXBException;
+import java.util.HashMap;
 import org.hl7.cql_annotations.r1.CqlToElmBase;
 import org.hl7.elm.r1.Library;
 
 public class ElmJsonMapper {
 
     private static JAXBContext jaxbContext;
+    private static HashMap<String, String> properties = new HashMap<>();
+
+    static {
+        properties.put(JAXBContext.JAXB_CONTEXT_FACTORY, "org.eclipse.persistence.jaxb.JAXBContextFactory");
+    }
 
     public static JAXBContext getJaxbContext() {
         if (jaxbContext == null) {
             try {
-                jaxbContext = JAXBContext.newInstance(Library.class, CqlToElmBase.class);
+                jaxbContext = JAXBContext.newInstance(new Class<?>[] {Library.class, CqlToElmBase.class}, properties);
             } catch (JAXBException e) {
                 throw new RuntimeException("Error creating JAXBContext - " + e.getMessage());
             }

--- a/Src/java/elm-jaxb/src/main/java/org/cqframework/cql/elm/serializing/jaxb/ElmXmlMapper.java
+++ b/Src/java/elm-jaxb/src/main/java/org/cqframework/cql/elm/serializing/jaxb/ElmXmlMapper.java
@@ -2,17 +2,24 @@ package org.cqframework.cql.elm.serializing.jaxb;
 
 import jakarta.xml.bind.JAXBContext;
 import jakarta.xml.bind.JAXBException;
+import java.util.HashMap;
 import org.cqframework.cql.elm.IdObjectFactory;
 
 public class ElmXmlMapper {
 
     private static JAXBContext jaxbContext;
+    private static HashMap<String, String> properties = new HashMap<>();
+
+    static {
+        properties.put(JAXBContext.JAXB_CONTEXT_FACTORY, "org.eclipse.persistence.jaxb.JAXBContextFactory");
+    }
 
     public static JAXBContext getJaxbContext() {
         if (jaxbContext == null) {
             try {
-                jaxbContext =
-                        JAXBContext.newInstance(IdObjectFactory.class, org.hl7.cql_annotations.r1.ObjectFactory.class);
+                jaxbContext = JAXBContext.newInstance(
+                        new Class<?>[] {IdObjectFactory.class, org.hl7.cql_annotations.r1.ObjectFactory.class},
+                        properties);
             } catch (JAXBException e) {
                 throw new RuntimeException("Error creating JAXBContext - " + e.getMessage());
             }

--- a/Src/java/elm-jaxb/src/main/resources/org/cqframework/cql/elm/serializing/jaxb/jaxb.properties
+++ b/Src/java/elm-jaxb/src/main/resources/org/cqframework/cql/elm/serializing/jaxb/jaxb.properties
@@ -1,1 +1,0 @@
-jakarta.xml.bind.context.factory=org.eclipse.persistence.jaxb.JAXBContextFactory

--- a/Src/java/elm-jaxb/src/test/java/org/cqframework/cql/elm/serializing/jaxb/ElmJsonLibraryReaderTest.java
+++ b/Src/java/elm-jaxb/src/test/java/org/cqframework/cql/elm/serializing/jaxb/ElmJsonLibraryReaderTest.java
@@ -1,0 +1,18 @@
+package org.cqframework.cql.elm.serializing.jaxb;
+
+import static org.testng.Assert.assertNotNull;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import org.hl7.elm.r1.Library;
+import org.testng.annotations.Test;
+
+public class ElmJsonLibraryReaderTest {
+
+    @Test
+    public void testRead() throws IOException {
+        var reader = new ElmJsonLibraryReader();
+        Library library = reader.read(new ByteArrayInputStream("{\"library\" : { \"type\" : \"Library\"}}".getBytes()));
+        assertNotNull(library);
+    }
+}

--- a/Src/java/elm-jaxb/src/test/java/org/cqframework/cql/elm/serializing/jaxb/ElmXmlLibraryReaderTest.java
+++ b/Src/java/elm-jaxb/src/test/java/org/cqframework/cql/elm/serializing/jaxb/ElmXmlLibraryReaderTest.java
@@ -1,0 +1,19 @@
+package org.cqframework.cql.elm.serializing.jaxb;
+
+import static org.testng.Assert.assertNotNull;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import org.hl7.elm.r1.Library;
+import org.testng.annotations.Test;
+
+public class ElmXmlLibraryReaderTest {
+
+    @Test
+    public void testRead() throws IOException {
+        var reader = new ElmXmlLibraryReader();
+        Library library =
+                reader.read(new ByteArrayInputStream("<library xmlns=\"urn:hl7-org:elm:r1\"></library>".getBytes()));
+        assertNotNull(library);
+    }
+}

--- a/Src/java/elm-test/src/test/java/org/cqframework/cql/elm/ElmDeserializeTests.java
+++ b/Src/java/elm-test/src/test/java/org/cqframework/cql/elm/ElmDeserializeTests.java
@@ -151,7 +151,7 @@ public class ElmDeserializeTests {
             throws IOException, JAXBException {
         Library xmlLibrary = null;
         try {
-            xmlLibrary = new org.cqframework.cql.elm.serializing.jaxb.ElmXmlLibraryReader()
+            xmlLibrary = new org.cqframework.cql.elm.serializing.jaxb.ElmJsonLibraryReaderTest()
                     .read(new FileReader(path + "/" + xmlFileName));
         } catch (Exception e) {
             throw new IllegalArgumentException(
@@ -332,8 +332,8 @@ public class ElmDeserializeTests {
                     .read(new StringReader(jaxbXml));
             validateEmptyStringsTest(xmlLibrary);
 
-            xmlLibrary =
-                    new org.cqframework.cql.elm.serializing.jaxb.ElmXmlLibraryReader().read(new StringReader(jaxbXml));
+            xmlLibrary = new org.cqframework.cql.elm.serializing.jaxb.ElmJsonLibraryReaderTest()
+                    .read(new StringReader(jaxbXml));
             validateEmptyStringsTest(xmlLibrary);
 
             // xmlLibrary = new org.cqframework.cql.elm.serializing.jackson.ElmXmlLibraryReader().read(new
@@ -378,7 +378,7 @@ public class ElmDeserializeTests {
     private static Library deserializeXmlLibrary(String filePath) throws IOException {
         final InputStream resourceAsStream = ElmDeserializeTests.class.getResourceAsStream(filePath);
         assertNotNull(resourceAsStream);
-        return new org.cqframework.cql.elm.serializing.jaxb.ElmXmlLibraryReader().read(resourceAsStream);
+        return new org.cqframework.cql.elm.serializing.jaxb.ElmJsonLibraryReaderTest().read(resourceAsStream);
     }
 
     private static void verifySigLevels(Library library, LibraryBuilder.SignatureLevel expectedSignatureLevel) {

--- a/Src/java/elm-test/src/test/java/org/cqframework/cql/elm/ElmDeserializeTests.java
+++ b/Src/java/elm-test/src/test/java/org/cqframework/cql/elm/ElmDeserializeTests.java
@@ -151,7 +151,7 @@ public class ElmDeserializeTests {
             throws IOException, JAXBException {
         Library xmlLibrary = null;
         try {
-            xmlLibrary = new org.cqframework.cql.elm.serializing.jaxb.ElmJsonLibraryReaderTest()
+            xmlLibrary = new org.cqframework.cql.elm.serializing.jaxb.ElmJsonLibraryReader()
                     .read(new FileReader(path + "/" + xmlFileName));
         } catch (Exception e) {
             throw new IllegalArgumentException(
@@ -332,8 +332,8 @@ public class ElmDeserializeTests {
                     .read(new StringReader(jaxbXml));
             validateEmptyStringsTest(xmlLibrary);
 
-            xmlLibrary = new org.cqframework.cql.elm.serializing.jaxb.ElmJsonLibraryReaderTest()
-                    .read(new StringReader(jaxbXml));
+            xmlLibrary =
+                    new org.cqframework.cql.elm.serializing.jaxb.ElmJsonLibraryReader().read(new StringReader(jaxbXml));
             validateEmptyStringsTest(xmlLibrary);
 
             // xmlLibrary = new org.cqframework.cql.elm.serializing.jackson.ElmXmlLibraryReader().read(new
@@ -378,7 +378,7 @@ public class ElmDeserializeTests {
     private static Library deserializeXmlLibrary(String filePath) throws IOException {
         final InputStream resourceAsStream = ElmDeserializeTests.class.getResourceAsStream(filePath);
         assertNotNull(resourceAsStream);
-        return new org.cqframework.cql.elm.serializing.jaxb.ElmJsonLibraryReaderTest().read(resourceAsStream);
+        return new org.cqframework.cql.elm.serializing.jaxb.ElmJsonLibraryReader().read(resourceAsStream);
     }
 
     private static void verifySigLevels(Library library, LibraryBuilder.SignatureLevel expectedSignatureLevel) {

--- a/Src/java/elm-test/src/test/java/org/cqframework/cql/elm/ElmDeserializeTests.java
+++ b/Src/java/elm-test/src/test/java/org/cqframework/cql/elm/ElmDeserializeTests.java
@@ -151,7 +151,7 @@ public class ElmDeserializeTests {
             throws IOException, JAXBException {
         Library xmlLibrary = null;
         try {
-            xmlLibrary = new org.cqframework.cql.elm.serializing.jaxb.ElmJsonLibraryReader()
+            xmlLibrary = new org.cqframework.cql.elm.serializing.jaxb.ElmXmlLibraryReader()
                     .read(new FileReader(path + "/" + xmlFileName));
         } catch (Exception e) {
             throw new IllegalArgumentException(
@@ -333,7 +333,7 @@ public class ElmDeserializeTests {
             validateEmptyStringsTest(xmlLibrary);
 
             xmlLibrary =
-                    new org.cqframework.cql.elm.serializing.jaxb.ElmJsonLibraryReader().read(new StringReader(jaxbXml));
+                    new org.cqframework.cql.elm.serializing.jaxb.ElmXmlLibraryReader().read(new StringReader(jaxbXml));
             validateEmptyStringsTest(xmlLibrary);
 
             // xmlLibrary = new org.cqframework.cql.elm.serializing.jackson.ElmXmlLibraryReader().read(new
@@ -378,7 +378,7 @@ public class ElmDeserializeTests {
     private static Library deserializeXmlLibrary(String filePath) throws IOException {
         final InputStream resourceAsStream = ElmDeserializeTests.class.getResourceAsStream(filePath);
         assertNotNull(resourceAsStream);
-        return new org.cqframework.cql.elm.serializing.jaxb.ElmJsonLibraryReader().read(resourceAsStream);
+        return new org.cqframework.cql.elm.serializing.jaxb.ElmXmlLibraryReader().read(resourceAsStream);
     }
 
     private static void verifySigLevels(Library library, LibraryBuilder.SignatureLevel expectedSignatureLevel) {

--- a/Src/java/elm-test/src/test/java/org/cqframework/cql/elm/ElmDeserializeTests.java
+++ b/Src/java/elm-test/src/test/java/org/cqframework/cql/elm/ElmDeserializeTests.java
@@ -336,13 +336,13 @@ public class ElmDeserializeTests {
                     new org.cqframework.cql.elm.serializing.jaxb.ElmXmlLibraryReader().read(new StringReader(jaxbXml));
             validateEmptyStringsTest(xmlLibrary);
 
-            // xmlLibrary = new org.cqframework.cql.elm.serializing.jackson.ElmXmlLibraryReader().read(new
-            // StringReader(jacksonXml));
-            // validateEmptyStringsTest(xmlLibrary);
+            Library jsonLibrary = new org.cqframework.cql.elm.serializing.jackson.ElmJsonLibraryReader()
+                    .read(new StringReader(jacksonJson));
+            validateEmptyStringsTest(jsonLibrary);
 
-            // xmlLibrary = new org.cqframework.cql.elm.serializing.jaxb.ElmXmlLibraryReader().read(new
-            // StringReader(jacksonXml));
-            // validateEmptyStringsTest(xmlLibrary);
+            jsonLibrary = new org.cqframework.cql.elm.serializing.jaxb.ElmJsonLibraryReader()
+                    .read(new StringReader(jaxbJson));
+            validateEmptyStringsTest(xmlLibrary);
         } catch (IOException e) {
             e.printStackTrace();
         }

--- a/Src/java/model-jaxb/src/main/java/org/hl7/elm_modelinfo/r1/serializing/jaxb/XmlModelInfoReader.java
+++ b/Src/java/model-jaxb/src/main/java/org/hl7/elm_modelinfo/r1/serializing/jaxb/XmlModelInfoReader.java
@@ -33,9 +33,14 @@ public class XmlModelInfoReader implements ModelInfoReader {
         return unmarshaller;
     }
 
+    @SuppressWarnings("unchecked")
+    private ModelInfo toModelInfo(Object obj) {
+        return ((JAXBElement<ModelInfo>) obj).getValue();
+    }
+
     public ModelInfo read(File src) throws IOException {
         try {
-            return ((JAXBElement<ModelInfo>) getUnmarshaller().unmarshal(src)).getValue();
+            return toModelInfo(getUnmarshaller().unmarshal(src));
         } catch (JAXBException e) {
             throw new IOException(e);
         }
@@ -43,7 +48,7 @@ public class XmlModelInfoReader implements ModelInfoReader {
 
     public ModelInfo read(Reader src) throws IOException {
         try {
-            return ((JAXBElement<ModelInfo>) getUnmarshaller().unmarshal(src)).getValue();
+            return toModelInfo(getUnmarshaller().unmarshal(src));
         } catch (JAXBException e) {
             throw new IOException(e);
         }
@@ -51,7 +56,7 @@ public class XmlModelInfoReader implements ModelInfoReader {
 
     public ModelInfo read(InputStream src) throws IOException {
         try {
-            return ((JAXBElement<ModelInfo>) getUnmarshaller().unmarshal(src)).getValue();
+            return toModelInfo(getUnmarshaller().unmarshal(src));
         } catch (JAXBException e) {
             throw new IOException(e);
         }
@@ -59,7 +64,7 @@ public class XmlModelInfoReader implements ModelInfoReader {
 
     public ModelInfo read(URL url) throws IOException {
         try {
-            return ((JAXBElement<ModelInfo>) getUnmarshaller().unmarshal(url)).getValue();
+            return toModelInfo(getUnmarshaller().unmarshal(url));
         } catch (JAXBException e) {
             throw new IOException(e);
         }
@@ -67,7 +72,7 @@ public class XmlModelInfoReader implements ModelInfoReader {
 
     public ModelInfo read(URI uri) throws IOException {
         try {
-            return ((JAXBElement<ModelInfo>) getUnmarshaller().unmarshal(uri.toURL())).getValue();
+            return toModelInfo(getUnmarshaller().unmarshal(uri.toURL()));
         } catch (JAXBException e) {
             throw new IOException(e);
         }
@@ -75,8 +80,7 @@ public class XmlModelInfoReader implements ModelInfoReader {
 
     public ModelInfo read(String string) throws IOException {
         try {
-            return ((JAXBElement<ModelInfo>) getUnmarshaller().unmarshal(new ByteArrayInputStream(string.getBytes())))
-                    .getValue();
+            return toModelInfo(getUnmarshaller().unmarshal(new ByteArrayInputStream(string.getBytes())));
         } catch (JAXBException e) {
             throw new IOException(e);
         }

--- a/Src/java/model-jaxb/src/main/java/org/hl7/elm_modelinfo/r1/serializing/jaxb/XmlModelInfoReader.java
+++ b/Src/java/model-jaxb/src/main/java/org/hl7/elm_modelinfo/r1/serializing/jaxb/XmlModelInfoReader.java
@@ -1,6 +1,7 @@
 package org.hl7.elm_modelinfo.r1.serializing.jaxb;
 
 import jakarta.xml.bind.JAXBContext;
+import jakarta.xml.bind.JAXBElement;
 import jakarta.xml.bind.JAXBException;
 import jakarta.xml.bind.Unmarshaller;
 import java.io.*;
@@ -9,6 +10,7 @@ import java.net.URL;
 import java.util.HashMap;
 import java.util.Map;
 import org.hl7.elm_modelinfo.r1.ModelInfo;
+import org.hl7.elm_modelinfo.r1.ObjectFactory;
 import org.hl7.elm_modelinfo.r1.serializing.ModelInfoReader;
 
 public class XmlModelInfoReader implements ModelInfoReader {
@@ -20,7 +22,7 @@ public class XmlModelInfoReader implements ModelInfoReader {
         properties.put(JAXBContext.JAXB_CONTEXT_FACTORY, "org.eclipse.persistence.jaxb.JAXBContextFactory");
 
         try {
-            this.unmarshaller = JAXBContext.newInstance(new Class<?>[] {ModelInfo.class}, properties)
+            this.unmarshaller = JAXBContext.newInstance(new Class<?>[] {ObjectFactory.class}, properties)
                     .createUnmarshaller();
         } catch (JAXBException e) {
             throw new RuntimeException(e);
@@ -33,7 +35,7 @@ public class XmlModelInfoReader implements ModelInfoReader {
 
     public ModelInfo read(File src) throws IOException {
         try {
-            return (ModelInfo) getUnmarshaller().unmarshal(src);
+            return ((JAXBElement<ModelInfo>) getUnmarshaller().unmarshal(src)).getValue();
         } catch (JAXBException e) {
             throw new IOException(e);
         }
@@ -41,7 +43,7 @@ public class XmlModelInfoReader implements ModelInfoReader {
 
     public ModelInfo read(Reader src) throws IOException {
         try {
-            return (ModelInfo) getUnmarshaller().unmarshal(src);
+            return ((JAXBElement<ModelInfo>) getUnmarshaller().unmarshal(src)).getValue();
         } catch (JAXBException e) {
             throw new IOException(e);
         }
@@ -49,7 +51,7 @@ public class XmlModelInfoReader implements ModelInfoReader {
 
     public ModelInfo read(InputStream src) throws IOException {
         try {
-            return (ModelInfo) getUnmarshaller().unmarshal(src);
+            return ((JAXBElement<ModelInfo>) getUnmarshaller().unmarshal(src)).getValue();
         } catch (JAXBException e) {
             throw new IOException(e);
         }
@@ -57,7 +59,7 @@ public class XmlModelInfoReader implements ModelInfoReader {
 
     public ModelInfo read(URL url) throws IOException {
         try {
-            return (ModelInfo) getUnmarshaller().unmarshal(url);
+            return ((JAXBElement<ModelInfo>) getUnmarshaller().unmarshal(url)).getValue();
         } catch (JAXBException e) {
             throw new IOException(e);
         }
@@ -65,7 +67,7 @@ public class XmlModelInfoReader implements ModelInfoReader {
 
     public ModelInfo read(URI uri) throws IOException {
         try {
-            return (ModelInfo) getUnmarshaller().unmarshal(uri.toURL());
+            return ((JAXBElement<ModelInfo>) getUnmarshaller().unmarshal(uri.toURL())).getValue();
         } catch (JAXBException e) {
             throw new IOException(e);
         }
@@ -73,7 +75,8 @@ public class XmlModelInfoReader implements ModelInfoReader {
 
     public ModelInfo read(String string) throws IOException {
         try {
-            return (ModelInfo) getUnmarshaller().unmarshal(new ByteArrayInputStream(string.getBytes()));
+            return ((JAXBElement<ModelInfo>) getUnmarshaller().unmarshal(new ByteArrayInputStream(string.getBytes())))
+                    .getValue();
         } catch (JAXBException e) {
             throw new IOException(e);
         }

--- a/Src/java/model-jaxb/src/main/java/org/hl7/elm_modelinfo/r1/serializing/jaxb/XmlModelInfoReader.java
+++ b/Src/java/model-jaxb/src/main/java/org/hl7/elm_modelinfo/r1/serializing/jaxb/XmlModelInfoReader.java
@@ -1,35 +1,81 @@
 package org.hl7.elm_modelinfo.r1.serializing.jaxb;
 
-import jakarta.xml.bind.JAXB;
+import jakarta.xml.bind.JAXBContext;
+import jakarta.xml.bind.JAXBException;
+import jakarta.xml.bind.Unmarshaller;
 import java.io.*;
 import java.net.URI;
 import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
 import org.hl7.elm_modelinfo.r1.ModelInfo;
 import org.hl7.elm_modelinfo.r1.serializing.ModelInfoReader;
 
 public class XmlModelInfoReader implements ModelInfoReader {
 
+    private final Map<String, String> properties = new HashMap<>();
+    private final Unmarshaller unmarshaller;
+
+    public XmlModelInfoReader() {
+        properties.put(JAXBContext.JAXB_CONTEXT_FACTORY, "org.eclipse.persistence.jaxb.JAXBContextFactory");
+
+        try {
+            this.unmarshaller = JAXBContext.newInstance(new Class<?>[] {ModelInfo.class}, properties)
+                    .createUnmarshaller();
+        } catch (JAXBException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Unmarshaller getUnmarshaller() {
+        return unmarshaller;
+    }
+
     public ModelInfo read(File src) throws IOException {
-        return JAXB.unmarshal(src, ModelInfo.class);
+        try {
+            return (ModelInfo) getUnmarshaller().unmarshal(src);
+        } catch (JAXBException e) {
+            throw new IOException(e);
+        }
     }
 
     public ModelInfo read(Reader src) throws IOException {
-        return JAXB.unmarshal(src, ModelInfo.class);
+        try {
+            return (ModelInfo) getUnmarshaller().unmarshal(src);
+        } catch (JAXBException e) {
+            throw new IOException(e);
+        }
     }
 
     public ModelInfo read(InputStream src) throws IOException {
-        return JAXB.unmarshal(src, ModelInfo.class);
+        try {
+            return (ModelInfo) getUnmarshaller().unmarshal(src);
+        } catch (JAXBException e) {
+            throw new IOException(e);
+        }
     }
 
     public ModelInfo read(URL url) throws IOException {
-        return JAXB.unmarshal(url, ModelInfo.class);
+        try {
+            return (ModelInfo) getUnmarshaller().unmarshal(url);
+        } catch (JAXBException e) {
+            throw new IOException(e);
+        }
     }
 
     public ModelInfo read(URI uri) throws IOException {
-        return JAXB.unmarshal(uri, ModelInfo.class);
+        try {
+            return (ModelInfo) getUnmarshaller().unmarshal(uri.toURL());
+        } catch (JAXBException e) {
+            throw new IOException(e);
+        }
     }
 
     public ModelInfo read(String string) throws IOException {
-        return JAXB.unmarshal(string, ModelInfo.class);
+        try {
+            return (ModelInfo) getUnmarshaller().unmarshal(new ByteArrayInputStream(string.getBytes()));
+        } catch (JAXBException e) {
+            throw new IOException(e);
+        }
     }
 }

--- a/Src/java/model-jaxb/src/main/resources/java/org/hl7/elm_modelinfo/r1/serializing/jaxb/jaxb.properties
+++ b/Src/java/model-jaxb/src/main/resources/java/org/hl7/elm_modelinfo/r1/serializing/jaxb/jaxb.properties
@@ -1,1 +1,0 @@
-jakarta.xml.bind.context.factory=org.eclipse.persistence.jaxb.JAXBContextFactory

--- a/Src/java/model-jaxb/src/test/java/org/hl7/elm_modelinfo/r1/serializing/jaxb/XmlModelInfoReaderTest.java
+++ b/Src/java/model-jaxb/src/test/java/org/hl7/elm_modelinfo/r1/serializing/jaxb/XmlModelInfoReaderTest.java
@@ -1,0 +1,18 @@
+package org.hl7.elm_modelinfo.r1.serializing.jaxb;
+
+import static org.testng.Assert.assertNotNull;
+
+import java.io.IOException;
+import org.hl7.elm_modelinfo.r1.ModelInfo;
+import org.testng.annotations.Test;
+
+public class XmlModelInfoReaderTest {
+
+    @Test
+    public void testRead() throws IOException {
+        var reader = new XmlModelInfoReader();
+        ModelInfo mi = reader.read(
+                "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?><ns4:modelInfo xmlns:xs=\"http://www.w3.org/2001/XMLSchema\" xmlns:ns4=\"urn:hl7-org:elm-modelinfo:r1\" xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"></ns4:modelInfo>");
+        assertNotNull(mi);
+    }
+}

--- a/Src/java/model/src/main/resources/org/hl7/elm_modelinfo/r1/jaxb.properties
+++ b/Src/java/model/src/main/resources/org/hl7/elm_modelinfo/r1/jaxb.properties
@@ -1,1 +1,0 @@
-jakarta.xml.bind.context.factory=org.eclipse.persistence.jaxb.JAXBContextFactory

